### PR TITLE
Create up down key press functionality

### PR DESCRIPTION
**Pull request by RingoftheKing**
Now the up and down keys scroll through the user's command history.

Limited to the current use session only

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -58,11 +62,13 @@ public void init() throws Exception {
         UserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(config.getUserPrefsFilePath());
         UserPrefs userPrefs = initPrefs(userPrefsStorage);
         AddressBookStorage addressBookStorage = new JsonAddressBookStorage(userPrefs.getAddressBookFilePath());
-        storage = new StorageManager(addressBookStorage, userPrefsStorage);
+        CommandHistoryStorage commandHistoryStorage = new JsonCommandHistoryStorage(config.getCommandHistoryFilePath());
+        storage = new StorageManager(addressBookStorage, userPrefsStorage, commandHistoryStorage);
+        CommandHistory c = initCommandHistory(commandHistoryStorage);
```
Comment (Ragnapop) [inline src/main/java/seedu/address/MainApp.java:67]: possibly could change the name of the CommandHistory

File: src/main/java/seedu/address/logic/CommandHistory.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+     * Adds a command to the Command History of the user.
+     * Removes the oldest command if the length exceeds 50
+     * @param s the Command string to store
+     */
+    public void add(String s) {
+        internalList.add(s);
+
+        // Ensures the list doesn't grow infinitely
+        if (internalList.size() > 50) {
+            internalList.remove(0);
```
Comment (Ragnapop) [inline src/main/java/seedu/address/logic/CommandHistory.java:51]: very interesting case that was addressed

File: src/main/java/seedu/address/logic/CommandHistory.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    /**
+     * Returns the next Command relative to the cursor's current position
+     */
+    public Optional<String> getNextCommand() {
+        if (historyIsEmpty()) {
+            return Optional.empty();
+        }
+
+        if (cursorPosition < internalList.size() - 1) {
```
Comment (Ragnapop) [inline src/main/java/seedu/address/logic/CommandHistory.java:78]: addressed if cursorPosition iterates outside of internalList size

**Review by Ragnapop — COMMENTED** [review]
LGTM!

**Comment by hjuntan** [comment]
LGTM

**Pull request merged**